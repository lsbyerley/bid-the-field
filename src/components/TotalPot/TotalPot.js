import { getTotalPot } from '../../lib/auctionUtils';
import useAsyncReference from '../../lib/useAsyncReference';

const TotalPot = ({ bids = [] }) => {
  const asyncBids = useAsyncReference(bids, true);
  const totalPotAmount = getTotalPot(asyncBids.current);

  return (
    <div className='mb-5 rounded-lg card card-compact bg-base-200'>
      <div className='card-body'>
        <h3 className='text-lg font-medium'>
          <span className='mr-2 text-sm'>Total Pot:</span>
          <span className='text-success'>
            ${Number.parseFloat(totalPotAmount).toFixed(2)}
          </span>
        </h3>
      </div>
    </div>
  );
};

export default TotalPot;
