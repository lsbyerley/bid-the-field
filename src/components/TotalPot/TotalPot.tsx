import { getTotalPot } from '@/lib/auctionUtils';
import useAsyncReference from '@/lib/useAsyncReference';

import { TotalPotArgs } from '@/types';

const TotalPot = ({ bids = [] }: TotalPotArgs) => {
  const asyncBids = useAsyncReference(bids, true);
  const totalPotAmount = getTotalPot(asyncBids.current);

  return (
    <div className='rounded-lg card card-compact bg-base-100'>
      <div className='items-center justify-center card-body'>
        <h3 className='text-lg font-medium'>
          <span className='mr-2 text-sm'>Total Pot:</span>
          <span className='text-success'>${totalPotAmount.toFixed(2)}</span>
        </h3>
      </div>
    </div>
  );
};

export default TotalPot;
