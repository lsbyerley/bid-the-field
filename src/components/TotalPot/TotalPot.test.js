import React from 'react';
import { render, screen } from 'test-utils';
import TotalPot from './TotalPot';

const defaultProps = (overrides) => {
  return {
    bids: [
      {
        id: 'test',
        owner: 'testowner',
        amount: '25',
        player_id: 'test',
        auction_id: 'test',
      },
    ],
    ...overrides,
  };
};

const renderTotalPot = (props) => {
  const utils = render(<TotalPot {...props} />);
  const home = utils.container.firstChild;
  return { ...utils, home };
};

describe('TotalPot', () => {
  it('should render and match snapshot', () => {
    const { home } = renderTotalPot(defaultProps());
    expect(home).toBeTruthy();
    expect(home).toMatchSnapshot();
  });
});
