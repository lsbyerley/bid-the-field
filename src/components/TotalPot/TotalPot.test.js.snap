// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalPot should render and match snapshot 1`] = `
<div
  class="rounded-lg card card-compact bg-base-100"
>
  <div
    class="items-center justify-center card-body"
  >
    <h3
      class="text-lg font-medium"
    >
      <span
        class="mr-2 text-sm"
      >
        Total Pot:
      </span>
      <span
        class="text-success"
      >
        $
        25.00
      </span>
    </h3>
  </div>
</div>
`;
